packages/next-codemod/bin/upgrade.ts

```diff
@@ -611,3 +617,89 @@ function writeOverridesField(
     }
   }
 }
+
+function warnDependenciesOutOfRange(
+  appPackageJson: any,
+  versionMapping: Record<string, { version: string; required: boolean }>
+) {
+  const allDependenciesToInstall = {
+    ...appPackageJson.dependencies,
+    ...appPackageJson.devDependencies,
+  }
+
+  const dependenciesOutOfRange = new Map<
+    string,
+    {
+      [dependency: string]: {
+        currentVersion: string
+        expectedVersionRange: string
+      }
+    }
+  >()
+
+  for (const dependency of Object.keys(allDependenciesToInstall)) {
+    let pkgJson
+    try {
+      pkgJson = require(
+        require.resolve(`${dependency}/package.json`, { paths: [cwd] })
+      )
+    } catch (error) {
+      if (error.code === 'ERR_PACKAGE_PATH_NOT_EXPORTED') {
+        pkgJson = JSON.parse(
+          fs.readFileSync(
+            path.join(cwd, 'node_modules', dependency, 'package.json'),
+            'utf8'
+          )
+        )
```

Comment:
This needs to be in try-catch as well in case the package manager uses a different node_modules layout

Reply:
In that case, what would we do in catch?

Reply:
warn and continue